apps/livreGUI/Controller.cpp

```diff
@@ -21,45 +21,159 @@
  */
 
 
-#include "Controller.h"
-#include <zeq/hbp/hbp.h>
+#include <livreGUI/Controller.h>
+
+#include <zeq/subscriber.h>
+#include <zeq/publisher.h>
+#include <zeq/event.h>
+
+#include <zeq/hbp/vocabulary.h>
+
+#include <isc/steering/simulator.h>
+
+#include <boost/thread.hpp>
 
 namespace livre
 {
 
+struct Controller::Impl
+{
+
+    typedef std::map< std::string, zeq::Publisher* > PublisherMap;
+    typedef std::pair< std::string, zeq::Subscriber* > URISubscriberPair;
+    typedef std::map< std::string, zeq::Subscriber* > SubscriberMap;
+    typedef std::map< std::string, isc::Simulator* > SimulatorMap;
+
+    Impl()
+        : _transferFunction( new TransferFunction1Df())
+        , _subscriberPoll( boost::bind( &Impl::pollSubscribers, this ))
+        , _continuePolling( true )
+
+    {}
+
+    ~Impl()
+    {
+        _continuePolling = false;
+        _subscriberPoll.join();
+    }
+
+    bool connect( const std::string&, const uint16_t )
+    {
+        LBUNIMPLEMENTED;
+        return false;
+    }
+
+    std::string getURIStr( const servus::URI& uri )
+    {
+        std::stringstream ss;
+        ss << uri;
+        return ss.str();
+    }
+
+    template< typename Type >
+    Type* getObject( std::map< std::string, Type* >& map,
+                const servus::URI& uri  )
+    {
+        const std::string& uriStr = getURIStr( uri );
+        typename std::map< std::string, Type* >::iterator it = map.find( uriStr );
+        if( it == map.end( ))
+            map[ uriStr ] = new Type( uri );
```

Comment:
No corresponding delete? ~impl should properly destroy allocated publishers and subscribers.